lxd/patches.go

```diff
@@ -168,6 +169,71 @@ func patchesApply(d *Daemon, stage patchStage) error {
 
 // Patches begin here
 
+// patchThinpoolTypoFix renames any config incorrectly set config file entries due to the lvm.thinpool_name typo.
+func patchThinpoolTypoFix(name string, d *Daemon) error {
+	err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		// Getting all storage pools.
+		pools, err := d.cluster.GetStoragePoolNames()
+		if err != nil {
+			if err == db.ErrNoSuchObject {
+				return nil
+			}
+
+			return err
+		}
+
+		// Getting all nodes.
+		nodes, err := tx.GetNodes()
+		if err != nil {
+			return err
+		}
+
+		// Update keys for each storage pool.
+		for _, poolName := range pools {
+			poolID, pool, err := d.cluster.GetStoragePoolInAnyState(poolName)
+			if err != nil {
+				return err
+			}
+
+			// Get the config of this storage pool.
+			config, err := tx.getStoragePoolConfig(poolID)
+			if err != nil {
+				return err
+			}
+
+			// See if we have the key we're looking for.
+			value, ok := config["lvm.thinpool_name"]
+			if !ok {
+				continue
+			}
+
+			// If we do, delete the current key
+			err = clearStoragePoolConfig(tx.tx, poolID, nil)
+			if err != nil {
+				return err
+			}
```

Comment:
You can't do that. For one thing, `clearStoragePoolConfig` is an unexported function so you can't access it.
For another thing, even if you could access it, this would wipe the ENTIRE pool configuration, not just the one key you're trying to remove.